amazonclouddrive/amazonclouddrive.go

```diff
@@ -597,6 +597,110 @@ func (f *Fs) Mkdir() error {
 	return f.dirCache.FindRoot(true)
 }
 
+// Move src to this remote using server side move operations.
+//
+// This is stored with the remote path given
+//
+// It returns the destination Object and a possible error
+//
+// Will only be called if src.Fs().Name() == f.Name()
+//
+// If it isn't possible then return fs.ErrorCantMove
+func (f *Fs) Move(src fs.Object, remote string) (fs.Object, error) {
+	srcObj, ok := src.(*Object)
+	if !ok {
+		fs.Debug(src, "Can't move - not same remote type")
+		return nil, fs.ErrorCantMove
+	}
+	fs.Debug(src, "\n\nTrying to move '%s' -> '%s'", srcObj.Remote(), remote)
+
+	// Temporary Object under construction
+	dstObj := &Object{
+		fs:     f,
+		remote: remote,
+	}
+	// Check if object already exists
+	err := dstObj.readMetaData()
+	switch err {
+	case nil:
+		// TODO: directory exists?
```

Comment:
I wonder why I didn't show up for you, I thought I had pushed the patch earlier. Either way, Amazon would just reply with 409 Conflict in those cases instead of overwriting. It would be messy though, so no more overwriting at all.